src/EnergyPlus/SolarCollectors.cc

```diff
@@ -1148,24 +1149,25 @@ namespace SolarCollectors {
             Real64 FRULpTest = 0.0;
 
             // Modify coefficients depending on test correlation type
-            {
-                auto const SELECT_CASE_var(state.dataSolarCollectors->Parameters(ParamNum).TestType);
-                if (SELECT_CASE_var == TestTypeEnum::INLET) {
-                    FRULpTest = state.dataSolarCollectors->Parameters(ParamNum).eff1 +
-                                state.dataSolarCollectors->Parameters(ParamNum).eff2 * (inletTemp - state.dataSurface->SurfOutDryBulbTemp(SurfNum));
-                    TestTypeMod = 1.0;
-
-                } else if (SELECT_CASE_var == TestTypeEnum::AVERAGE) {
-                    FRULpTest = state.dataSolarCollectors->Parameters(ParamNum).eff1 +
-                                state.dataSolarCollectors->Parameters(ParamNum).eff2 *
-                                    ((inletTemp + outletTemp) * 0.5 - state.dataSurface->SurfOutDryBulbTemp(SurfNum));
-                    TestTypeMod = 1.0 / (1.0 - FRULpTest / (2.0 * mCpATest));
-
-                } else if (SELECT_CASE_var == TestTypeEnum::OUTLET) {
-                    FRULpTest = state.dataSolarCollectors->Parameters(ParamNum).eff1 +
-                                state.dataSolarCollectors->Parameters(ParamNum).eff2 * (outletTemp - state.dataSurface->SurfOutDryBulbTemp(SurfNum));
-                    TestTypeMod = 1.0 / (1.0 - FRULpTest / mCpATest);
-                }
+            switch (state.dataSolarCollectors->Parameters(ParamNum).TestType) {
+            case TestTypeEnum::INLET: {
```

Comment:
Question for my own edification: what is the "right" place for the `break` statement? I've always put them inside of the curly braces but it probably doesn't really matter. Bueller... Bueller...

Reply:
The braces are technically not even necessary, although they are good hygiene, so it doesn’t matter.